barretenberg_wrapper/build.rs

```diff
@@ -33,14 +31,14 @@ fn select_arch() -> Arch {
     let arch = std::env::consts::ARCH;
     match arch {
         "arm" => Arch::Arm,
+        "aarch64" => Arch::Arm,
```

Comment:
This helps M2 Macs with returned architecture of `aarch64` (instead of `arm`) from `std::env::consts::ARCH` to pick the correct toolchain.

It prevents them running on the fallback x86_64 config that brings compilation involving assembly code, which would lead to the `unknown token in expression` error when ran on ARM (described in more details [here](https://github.com/AztecProtocol/barretenberg/issues/66)).